src/index.js

```diff
@@ -181,14 +191,21 @@ class Searcher extends NxusModule {
     this._searchDocuments = {}
     this._createSearchDocument()
 
+    // Global listeners
```

Comment:
This is the change to fix multiple listeners in constructor rather than registered with every call to `searchable`, although the handlers handle every model in one listen.

Reply:
I see. So if I happened to call `.searchable()` five times, the `_handle*` methods would end up getting registered five times.

Reply:
Yes. An equally valid fix would be to have the listeners be only tied to the the `model.create.<identity>` events, and go back to listening in `searchable`. I might do that - they're currently written to be global and only do something if the triggered model identity is `searchable`, but that does mean they fire (and then do nothing) for every model change (even e.g. `users` etc).

Reply:
So you'd listen for the model-specific events (I forget the details of what's available)? Under the principle of "fewest surprises," that makes sense. That making one model searchable then impacts operations on _other_ models seems undesirable (although I suppose the practical effect is pretty small).

Reply:
Yep, updated model handlers back to `searchable` with that change. Much more confident that kind of plumbing change is correct with these tests still passing!